cmd/cluster-network-operator/main.go

```diff
@@ -27,30 +31,25 @@ func printVersion() {
 
 const LOCK_NAME = "cluster-network-operator"
 
-// urlOnlyKubeconfig is a slight hack; we need to get the apiserver from the
-// kubeconfig but should use the in-cluster service account
-var urlOnlyKubeconfig string
-
-func init() {
-	flag.StringVar(&urlOnlyKubeconfig, "url-only-kubeconfig", "",
-		"Path to a kubeconfig, but only for the apiserver url.")
-}
+const ENV_URL_KUBECONFIG = "URL_ONLY_KUBECONFIG"
 
 func main() {
-	printVersion()
-	flag.Parse()
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	pflag.CommandLine.SetNormalizeFunc(utilflag.WordSepNormalizeFunc)
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
 
-	namespace := "" // non-namespaced
+	logs.InitLogs()
+	defer logs.FlushLogs()
 
-	// TODO: Expose metrics port after SDK uses controller-runtime's dynamic client
-	// sdk.ExposeMetricsPort()
+	command := newNetworkOperatorCommand()
 
 	// Hack: the network operator can't use the apiserver service ip, since there's
 	// no network. We also can't hard-code it to 127.0.0.1, because we run during
 	// bootstrap. Instead, we bind-mount in the kubelet's kubeconfig, but just
 	// use it to get the apiserver url.
-	if urlOnlyKubeconfig != "" {
-		kubeconfig, err := clientcmd.LoadFromFile(urlOnlyKubeconfig)
+	if kc := os.Getenv(ENV_URL_KUBECONFIG); kc != "" {
```

Comment:
if we are losing -url-only-kubeconfig CLI option, maybe document somewhere this new env var?

Reply:
Good point.